documentation/SOFTWARE_REQUIREMENTS_SPEC.md

```diff
@@ -53,8 +53,8 @@ Our web-based Sokoban game will have greater user reachability compared to a mor
 The game shall be run on any browser that supports WASM (Safari, Chrome, Firefox, Edge, Opera, IOS Safari, Firefox for Android)
 
 ### Use Cases
-* The user presses the start button to initiate the game. A prompt will display asking for the user to enter level. A static world elements will be displayed on the screen.
+* The user presses the start button to initiate the game. A prompt will display asking for the user to enter level. A static board elements will be displayed on the screen.
 * The user can press any directional keys to move around the board. When the user preses UP on the keyboard, the character will move one cell up if the cell is empty or if there is a box and the adjacent cell of the box is empty. The user uses the UP, DOWN, LEFT, RIGHT keys to navigate around until all of the boxes are places on marked goals. This will clear the level.
-* When all of the boxes are placed on the goal cells, a congratulatory message and stats will be displayed. The user will be asked to play again.
-* The user will be prompted to move to the next level. 
-* The user clicks the next button. A new board will be displayed.
+* When all of the boxes are placed on the goal cells, a congratulatory message will appear.
+* The user will be prompted to move to the next level or jump to a different level.
```

Comment:
Maybe automatically moving them directly to the next level and providing a menu option to change levels any time would be best, to avoid decision fatigue and keep the flow going.

Reply:
I like that. Maybe, we can have a count down that can automatically transition to the next level and also an option to click the NEXT (for eager players) and PLAY AGAIN (for players who want to redo the level.
![Screen Shot 2022-02-27 at 8 50 21 AM](https://user-images.githubusercontent.com/76274300/155895656-8da6e1ef-4a3f-449c-9295-b8018b0ba5fa.png)

Reply:
I think of puzzle games as relaxing, so I'd prefer a non-countdown -- avoiding presenting it to feel like a race or forced competition/gamification. Maybe it's just an older generation thing ;-).

Reply:
I was thinking of it as a heads up that the default setting for the game is an "auto-play" so we'll pause for a few seconds before transitioning to the next level and at the same time allowing our users to give them the time to choose to either play the same level or quickly move on to the next level.

Reply:
Instead of autoplay and a countdown, how about the screen shows a "level complete" message or something, then waits for the user to press a button to play the next level, or returns them to the level selection screen?

Reply:
We could also make autoplay toggle-able, maybe with the default off. I'll approve the PR and I think we can tweak this as needed later on. One thing to keep in mind is that anything with a timer can make Puppeteer testing a little bit slower/harder/more involved.

Reply:
Does this restrict js animations?

Reply:
Yes, in the sense that the extent to which we add animations moves the project closer to the "harder to test" zone we were originally trying to avoid, namely a real-time game with things moving nondeterministically around the screen. We could add some of these things, but the more we add, the more pain it is (maybe not a bad thing, and we don't need to test everything). I'm just erring towards simplicity for starters.

Puppeteer has tools to wait for things to occur, but if everything on the screen updates instantly, we don't need to use them and our tests will be easier to write, faster to run (everything is real-time, so if we have a 3 second animation that we test, that test will take 3 seconds longer to run every time) and less prone to errors that can show up trying to synchronize Puppeteer with the live app.